### PR TITLE
Clear result on transaction retry

### PR DESCRIPTION
**Pull request by jpslav**
A tweaked version of #50.  Thanks @karenc for the initial draft -- can you review this and merge, then check that exchange passes its specs when this version of lev is used?  Then we can release a new version of lev and update exchange.

### CODE REVIEW

**Comment by karenc** [comment]
With this version of lev, exchange passes the spec in openstax/exchange#49.  I can't merge this since I don't have write access for this repo.

File: spec/routine_spec.rb [hunk 1]
```diff
@@ -116,4 +116,41 @@ def exec
     end
   end
 
+  it 'does not mess up results on a transaction retry' do
+    # To get the transaction to retry, we need to raise an exception the first time
+    # through execution, after an output has been set in a nested routine and
+    # translated to the parent routine
+
+    stub_const 'RaiseTransactionIsolationConflict', Class.new
```
Comment (karenc) [inline spec/routine_spec.rb]: Can we change the name to `NestedRoutine` or something?  It doesn't raise the transaction isolation conflict error anymore :smile:

**Comment by karenc** [comment]
Thanks very much for doing it.  I was working on it yesterday after Dante looked at it, but couldn't get the outputs to turn into an array.

File: spec/routine_spec.rb (hunk 1 above)
Comment (jpslav) [inline spec/routine_spec.rb]: Totally.  Done.

**Comment by jpslav** [comment]
You have write access now @karenc

**Pull request merged**